File: desktop/native/SurveyController.App/ViewModels/IPUsageRow.cpp
#include "pch.h"
#include "IPUsageRow.h"

#if __has_include("IPUsageRow.g.cpp")
#include "IPUsageRow.g.cpp"
#endif

namespace winrt::SurveyController::App::implementation
{
    IPUsageRow::IPUsageRow(hstring const& label, double total, double maximum)
        : m_label(label), m_total(total), m_maximum(maximum > 0 ? maximum : 1)
    {
    }
}

File: desktop/native-wpf/src/SurveyController.App/ViewModels/IpUsageRow.cs
namespace SurveyController.App.ViewModels;

/// <summary>IP 用量列表行：展示 DTO，创建后不可变。</summary>
public sealed class IpUsageRow
{
    public IpUsageRow(string label, double total, double maximum)
    {
        Label = label;
        Total = total;
        Maximum = maximum < 1 ? 1 : maximum;
    }

    public string Label { get; }

    public double Total { get; }

    public double Maximum { get; }

    public string CountText => $"{(int)Total} 个";
}
